src/utils/getUrlByRequestOptions.ts

```diff
@@ -50,15 +50,6 @@ function getPortByRequestOptions(
     return Number(options.port)
   }
 
-  // Extract the port from the hostname.
-  if (options.hostname != null) {
```

Comment:
Should this be `options.host` instead?

Reply:
`host` or `hostname` must not include port.

Reply:
🤔 What kind of hackery were we supporting with this. 

Ditto, let's remove it as well. Fewer hacks, everyone wins.